akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -55,7 +57,7 @@ object HttpCookiePair {
  * for a full definition of the http cookie header fields, see
  * http://tools.ietf.org/html/rfc6265
  */
-final case class HttpCookie(
+final class HttpCookie(
```

Comment:
Let's make the new constructor `private[http]` as well.

Reply:
Making it `private[http]` here increases the number of binary compatibility issues (since the constructor deprecated constructor does not have the default values). Source compatibility will be good though.

WDYT?

Reply:
I think it should be private. Couldn't we just move the default params to the deprecated constructor? In any case, we want people to use the apply method in the future.